mne/io/edf/edf.py

```diff
@@ -233,14 +246,18 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
                     stim[n_start:n_stop] += evid
                 data[stim_channel_idx, :] = stim[start:stop]
             else:
-                # Allows support for up to 17-bit trigger values (2 ** 17 - 1)
                 stim = np.bitwise_and(data[stim_channel_idx].astype(int),
-                                      131071)
+                                      2**17 - 1)
+
+                if biosemi:
+                    stim = stim >> 8  # not sure why but it works
```

Comment:
Ok I'm having second thoughts now. 

> I think I'd rather put this in a Notes section about dealing with BioSemi data than modify users' data.

The current GDF code in edf.py already modifies users' data, as the trigger values come out correct for bdf files (i.e. bits 16-23 of the status channel are discarded)

```
ch_data = np.fromfile(fid, dtype=dtype, count=samp * dtype_byte)
ch_data = ch_data.reshape(-1, 3).astype(np.int32)
ch_data = ((ch_data[:, 0]) +
            (ch_data[:, 1] << 8) +
            (ch_data[:, 2] << 16))
# 24th bit determines the sign
ch_data[ch_data >= (1 << 23)] -= (1 << 24)
```